cmd/ipfs/ulimit_test.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+	"testing"
+)
+
+func TestManagefd(t *testing.T) {
+	t.Log("Testing file descriptor count")
+	if err := managefd(); err != nil {
+		t.Errorf("Cannot manage file descriptors")
+	}
+
+	if maxFds != uint64(1024) {
+		t.Errorf("Maximum file descriptors default value changed")
+	}
+}
+
+func TestManageInvalidNFds(t *testing.T) {
+	t.Logf("Testing file descriptor invalidity")
+	var err error
+	if err = os.Unsetenv("IPFS_FD_MAX"); err != nil {
+		t.Errorf("Cannot unset the IPFS_FD_MAX env variable")
```

Comment:
Use fatal for these, the rest of the test is invalid if this assertion (or the others) fails.